pkg/templates/kubeadm/v1beta3/kubeadm.go

```diff
@@ -78,17 +78,24 @@ var (
 )
 
 // NewConfig returns all required configs to init a cluster via a set of v1beta3 configs
-func NewConfig(s *state.State, host kubeoneapi.HostConfig) ([]runtime.Object, error) {
+func NewConfig(s *state.State, host kubeoneapi.HostConfig) (
+	initConfig,
+	joinConfig,
+	clusterConfig,
+	kubeletConfig,
+	kubeproxyConfig runtime.Object,
```

Comment:
I'm wondering if we should instead use a struct that contains all these objects.

Reply:
Totally, such a big number of return values reads like a mess.

Reply:
@kron4eg Refactored into structs.